plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/TikaImpl.java

```diff
@@ -161,6 +161,8 @@ static PermissionCollection getRestrictedPermissions() {
         perms.add(new ReflectPermission("suppressAccessChecks"));
         // xmlbeans, use by POI, needs to get the context classloader
         perms.add(new RuntimePermission("getClassLoader"));
+        // ZipFile needs accessDeclaredMembers on Java 10
+        perms.add(new RuntimePermission("accessDeclaredMembers"));
```

Comment:
I think that we should have left an assertion here that the Java version is not JDK 11 (I think we will be able to remove this for JDK 11). I also think that this code should have been guarded by an if block checking that we are on JDK 10 and otherwise not add this permission.

Reply:
Of course. I can add the conditional check. Whats the best (in terms of least changing) system property used for this that is the same across different JVMs? Maybe "java.specification.version" or a prefix of "java.version"?
About adding an assertion, are we sure that this is going to change back in JDK 11? Would an issue to trask this be better?

Reply:
I opened #28603.